homeassistant/components/tolo/select.py

```diff
@@ -14,36 +17,77 @@
 from .const import DOMAIN
 
 
+@dataclass(frozen=True, kw_only=True)
+class ToloSelectEntityDescription(SelectEntityDescription):
+    """Class describing TOLO select entities."""
+
+    options: list[str]
+    getter: Callable[[ToloSettings], str]
+    setter: Callable[[ToloClient, str], bool]
+
+
+SELECTS = (
+    ToloSelectEntityDescription(
+        key="lamp_mode",
+        translation_key="lamp_mode",
+        options=[lamp_mode.name.lower() for lamp_mode in LampMode],
```

Comment:
I'd suggest hardcoding the options to avoid the risk of adding options that we don't have translations for. We want the integration to be in control of these items.

Reply:
Not exactly sure which change you are suggestion. Regardless of the technical implementation, I would suggest to have the names for the `LampMode` enum translated for the UI, to reflect "manual" and "automatic" in the respective language.

Reply:
He means, don't iterate over `LampMode` but instead just return a list `["manual", "automatic"]`

Reply:
That was what I first was doing, but this is how it evolves:

  * Instead of having plain strings in this list, strings should be moved to `.const` module and used here, so it becomes `[LAMP_MODE_MANUAL, LAMP_MODE_AUTOMATIC`].
  * The native way of "grouping" values into a limited set of possible choices is using enums, therefore I introduced the enum instead of just a bunch of variables.

I personally don't like the `.lower()` part in my solution, but using an enum for possible choices seems to be the best solution in my developers experience, as it is, per definition, a closed set of supported values for a specific purpose.

Anyway, you're project lead, if you want me to change that I will change that.

Reply:
It's ok to use a enum, but then that should be defined in the integration and not in the library.